src/main/java/org/jitsi/videobridge/AbstractEndpointMessageTransport.java

```diff
@@ -390,9 +392,17 @@ protected void onSelectedEndpointsChangedEvent(
      * {@code PinnedEndpointChangedEvent} which has been received.
      * @param newPinnedEndpoints the new pinned endpoints
      */
-    protected abstract void onPinnedEndpointsChangedEvent(
+    protected void onPinnedEndpointsChangedEvent(
         JSONObject jsonObject,
-        Set<String> newPinnedEndpoints);
+        Set<String> newPinnedEndpoints)
+    {
+        // Note that this captures the set.
+        pinnedEndpointConstraintsStream = newPinnedEndpoints
```

Comment:
Are you talking about removing the memory of what's pinned and what's selected? We need that, consider, for example, the case where a pinned endpoint comes on-stage and then off-stage again. It's still a pinned endpoint (the video constraints class has good info on what's pinned and what's selected in case it's confusing).

Reply:
I'll mark this conversation as resolved because we achieved this with the compatibility layer.